infra/core/networking/vnet.bicep

```diff
@@ -0,0 +1,73 @@
+param location string
+param name string
+param tags object = {}
+param appServicePlanName string
+param appServicePlanId string
+
+var addressPrefix = '10.0.0.0/16'
+
+var subnets = [
+  {
+    name: 'ai-subnet'
+    properties: {
+      addressPrefix:'10.0.1.0/24'
+      privateEndpointNetworkPolicies: 'Enabled'
+      privateLinkServiceNetworkPolicies: 'Enabled'      
+    }
+  }
+  {
+    name: 'AzureBastionSubnet'
+    properties: {
+      addressPrefix:'10.0.2.0/24'
+      privateEndpointNetworkPolicies: 'Enabled'
+      privateLinkServiceNetworkPolicies: 'Enabled'      
+    }
+  }
+  {
+    name: 'app-int-subnet'
```

Comment:
can we get consistent names?  I see pascal, dashes, etc

Reply:
I wasn't sure if AzureBastionSubnet was a magic name of some sort. @tonybaloney Can you comment? If its not a magic string, then we can use dashes for all of them. I felt like I saw a similar string in an example for AzureFrontDoor.

Reply:
I think it's a magic string. https://learn.microsoft.com/en-us/azure/bastion/bastion-overview#architecture